lib/jekyll/collection.rb

```diff
@@ -208,7 +208,7 @@ def read_document(full_path)
       doc = Jekyll::Document.new(full_path, :site => site, :collection => self)
       doc.read
       if site.publisher.publish?(doc) || !write?
```

Comment:
Why doesn't `publish?` check `hidden_in_the_future?`?

Reply:
> Why doesn't `publish?` check `hidden_in_the_future?` ?

@parkr, `:publish?` does `check hidden_in_the_future?` but it gets overridden by the `|| !write?`
`publisher == false` **but** `!write? == true` and therefore, the expression returns `true`

I initially thought about removing the `|| !write?` but changed it for some reason I do not remember now..